### DISCUSSION TITLE
What is the interval (sampling rate) between adjacent timestamps in csv records?

### DISCUSSION THREAD

**Discussion by RayShing** [nicolargo/glances#2095] [Q&A] [answered]
Hi, I would like to ask what is the interval (sampling rate) between adjacent timestamps (the first column) in the recording csv files? I found it is 2s for most of the time, and sometimes it changes to 3s. So what is the sampling rate for each recording?

**Comment by nicolargo** [comment]
Hi @felixshing 

Glances do not have sampling rate.

The minimal refresh rate is configure in the glances.conf file, it is the minimum duration between two samples.

The duration is variable and depend of the time take by Glances to grab all the stats.

Nicolas

**Comment by nicolargo** [comment, marked as answer]
To help you compute rate, every foo stats includes a foo_time_since_update field with the time between last update in second:

![image](https://user-images.githubusercontent.com/776747/180799773-88b795b0-a7ea-4a4f-ab77-d4543e41b765.png)

**Reply by RayShing** [reply]
Awesome! That is very helpful! Thank you very much!
